### PR TITLE
THEEDGE-1628: Change Fatal()/panic() to LogErrorandPanic() and lint f…

### PR DESCRIPTION
**Pull request by thom-at-redhat**
…ixes

# Description

Combine Fatal() call and panic() call into LogErrorandPanic() which flushes log messages.  Some lint fixes.  Stub ginkgo suite and test.

Fixes # THEEDGE-1628

## Type of change

What is it?

- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Documentation update
- [ ] Tests update
- [X] Refactor

# Checklist:

- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [X] New and existing unit tests pass locally with my changes
- [X] I run `go fmt ./...` to check that my code is properly formatted
- [X] I run `go vet ./...` to check that my code is free of common Go style mistakes

### CODE REVIEW

File: pkg/clients/client.go [hunk 1]
```diff
@@ -22,8 +21,7 @@ func GetOutgoingHeaders(ctx context.Context) map[string]string {
 		xhrid := identity.Get(ctx)
 		identityHeaders, err := json.Marshal(xhrid)
 		if err != nil {
-			log.WithField("error", err.Error()).Fatal("Error getting request ID")
-			panic(err)
+			logger.LogErrorandPanic("Error getting request ID", err)
```
Comment (Avielyo10) [inline pkg/clients/client.go:24]:
```suggestion
			logger.LogErrorAndPanic("Error getting request ID", err)
```

**Review by Avielyo10 — CHANGES_REQUESTED** [review]
Can you please omit the `go.*` files since they are not related to the PR goal.

I do suggest splitting this into 2 PRs, one for the panic & the other for linting stuff.

File: pkg/clients/clients_test.go [hunk 2]
```diff
@@ -0,0 +1,10 @@
+package clients_test
+
+import (
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
```
[the remark this answers is not in the document]
Comment (thom-at-redhat) [reply pkg/clients/clients_test.go:5]: it's a stub created by ginkgo...commented out

**Comment by thom-at-redhat** [comment]
go.* reverted per Aviel's suggestion

**Comment by thom-at-redhat** [comment]
> Can you please omit the `go.*` files since they are not related to the PR goal.
> 
> I do suggest splitting this into 2 PRs, one for the panic & the other for linting stuff.

Go.* are now omitted.  If it's ok, will do splitting on next PR

**Review by Avielyo10 — APPROVED** [review]
lgtm

**Pull request merged**